app/src/main/java/org/oppia/app/player/state/StatePlayerRecyclerViewAssembler.kt

```diff
@@ -443,40 +485,85 @@ class StatePlayerRecyclerViewAssembler private constructor(
 
   private fun maybeAddNavigationButtons(
     pendingItemList: MutableList<StateItemViewModel>,
+    rightPendingItemList: MutableList<StateItemViewModel>,
     hasPreviousState: Boolean,
     canContinueToNextState: Boolean,
     hasGeneralContinueButton: Boolean,
-    stateIsTerminal: Boolean
+    stateIsTerminal: Boolean,
+    shouldSplit: Boolean
   ) {
     val hasPreviousButton = playerFeatureSet.backwardNavigation && hasPreviousState
     when {
       hasGeneralContinueButton && playerFeatureSet.forwardNavigation -> {
-        pendingItemList += ContinueNavigationButtonViewModel(
-          hasPreviousButton,
-          previousNavigationButtonListener,
-          fragment as ContinueNavigationButtonListener
-        )
+        if (!shouldSplit) {
```

Comment:
There seems to be a lot of logical forks here. Can we maybe split this into a few different helpers to make it a bit easier to understand what is being setup when for each case?

Reply:
I'll try to figure out how to simplify it.  
For now, I have pushed the other suggested changes.

Reply:
I have tried.  
PTAL